src/Listing/Bid/Bid.model.js

```diff
@@ -81,12 +106,77 @@ export class Bid extends Model {
     // Re-validate bids for assets with its fingerprint back to the original value
     await this.db.query(
       SQL`UPDATE ${raw(this.tableName)}
-        SET status = ${LISTING_STATUS.open}
+        SET status = ${
+          LISTING_STATUS.open
+        }, block_time_updated_at = ${blockTime}, block_number = ${blockNumber}
         WHERE block_time_created_at <= ${blockTime}
           AND token_address = ${tokenAddress}
           AND token_id = ${tokenId}
           AND status = ${LISTING_STATUS.fingerprintChanged}
           AND fingerprint = ${fingerprint}`
     )
   }
+
+  static async updateAssetOwner(
+    tokenAddress,
+    assetId,
+    seller,
+    blockTime,
+    blockNumber
+  ) {
+    return this.db.query(
+      SQL`UPDATE ${raw(this.tableName)}
+        SET seller = ${seller}, block_time_updated_at = ${blockTime}, block_number = ${blockNumber}
+        WHERE block_time_created_at <= ${blockTime}
+          AND token_address = ${tokenAddress} 
+          AND asset_id = ${assetId}
+          AND status = ANY(${[
+            LISTING_STATUS.open,
+            LISTING_STATUS.fingerprintChanged
+          ]})`
+    )
+  }
```

Comment:
I think we should try to refactor the arguments of:
- invalidateBids
- updateAssetByFingerprintChange
- updateAssetOwner

The queries themselves seem ok, but those methods take 4+ arguments each and it's hard to distinguish what each of them do. Just adding an arg to the end of a function is usually a bad idea if you already have 3.

Maybe we can group a few of those into what they represent? Also we could try to keep the API of db.update where we have `(changes, conditions)`.

Plus I see a few repeated conditions like

```js
WHERE block_time_created_at <= ${blockTime}
  AND token_address = ${tokenAddress}
  AND token_id = ${tokenId}
```

but abstracting those might be going too far. I'll leave that decision to you

Reply:
Right, I will change the order and re-visit names